### PR TITLE
Add the first drafts of addon development docs

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
The first drafts of addon development docs. 

Some TODOs on my list: 
> It's also worth mentioning, that I can continue these TODOs in some follow up PRs.

- [ ] Add info for web actions
- [ ] Add More info about pyblish plugins (dev and debugging)
- [ ] any more suggestions ? 

> [!NOTE]
> This PR may take longer than usual to review.

## Additional info
These docs are the summary of many discussions and small experiments over the past year.
For a while, these notes remained in my notebook.
When I felt they were in a useful condition I thought of moving them here.
and, sorry for pushing a big commit.

## Testing notes:
Read through the docs and let me know what do you think.

### CODE REVIEW

File: website/docs/dev_addon_intro.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+:::info
+This page is still under construction 👷🚧🛠️🔜🏗️.
+
+Feel free to report any missing or wrong info.
+:::
+
+## Preface
+
+AYON is an studio platform for Animation, VFX and games.
```
Comment (mkolar) [inline website/docs/dev_addon_intro.md:15]: this is just repetition and feels like selling AYON even in the docs. let's be a bit more frugal with these :)

**Review by robin-ynput — APPROVED** [review]
Very meaningful !

**Review by Lypsolon — APPROVED** [review]
Very, very nice stuff like it. 
I wrote down a few things while reading the .md files. 
Feel free to ignore them if you want to.


I like that you start from addon-template 


dev_addon_creation.md

line: 89
Minimalist addon client definition
This code should exist in {my-addon}/client/my_addon/__init__.py
- is it the standard that the addon class lives in addon.py? 

line: 537
For more info about how to implement a host integration please fgo to Host implementation
I think there is an f too much in this sentence.

**Comment by MustafaJafar** [comment]
> line: 537
> For more info about how to implement a host integration please fgo to Host implementation
> I think there is an f too much in this sentence.

Creating new hosts can be fhard but, I think that's was typo 😂

**Comment by mkolar** [comment]
@MustafaJafar this is approved a couple of time, while actually being in draft....how come?

**Comment by MustafaJafar** [comment]
> @MustafaJafar this is approved a couple of time, while actually being in draft....how come?

@mkolar 
I'm still thinking of some additions like web actions. 
But, if you guys this PR is good as first draft. then lets merge and we can always improve it over time.

**Comment by mkolar** [comment]
I would merge and keep improving. If you think it's not complete, add a disclaimer on the top that it's still WIP, but I'de merge it

**Comment by MustafaJafar** [comment]
> I would merge and keep improving. If you think it's not complete, add a disclaimer on the top that it's still WIP, but I'de merge it


Good idea. I've just discovered that such a disclaimer already exists 😄 
Then let's merge.

**Pull request merged**